ffi/derive/src/ffi_fn.rs

```diff
@@ -153,20 +164,29 @@ fn gen_input_conversion_stmts(fn_descriptor: &FnDescriptor) -> TokenStream {
 fn gen_method_call_stmt(fn_descriptor: &FnDescriptor) -> TokenStream {
     let ident = &fn_descriptor.sig.ident;
     let self_type = &fn_descriptor.self_ty;
+    let trait_name = &fn_descriptor.trait_name;
 
     let receiver = fn_descriptor.receiver.as_ref();
     let self_arg_name = receiver.map_or_else(Vec::new, |arg| {
-        if matches!(arg.src_type(), syn::Type::Path(_)) {
+        if matches!(arg.src_type(), syn::Type::Path(_))
+            && Some(arg.name()) == fn_descriptor.output_arg.as_ref().map(Arg::name)
```

Comment:
```suggestion
        if Some(arg.name()) == fn_descriptor.output_arg.as_ref().map(Arg::name)
```

is this sufficient? It should not be possible to have two types with the same name, unless it's a special case for `fn(self) -> Self`.

Reply:
Yes, good point, `receiver` and `output_arg` will have same name only in this case.